libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/RouterWrapper.kt

```diff
@@ -183,12 +185,14 @@ class RouterWrapper(
             return REQUEST_FAILURE
         }
 
+        val indicesSnapshot = currentIndicesSnapshotProvider()!!
```

Comment:
This seems error-prone since the systems that interact with the `CurrentIndicesSnapshotProvider` are indirectly coupled.

What if the function provided to the `RouteWrapper` is implemented directly in the `RouteRefreshController`, and the controller is responsible for storing the indices? That would make the coupling direct (since it's required anyway).

Let's treat it as another idea for next iteration.

Reply:
Sounds reasonable, I like it.

Reply:
Ah no, that's the problem. If it were that simple, I'd just pass the indices. We can't add a parameter to `Router` method since it's public API. I'm gonna try Vadzim's solution tho.

Reply:
I'm not proposing adding a method to the router interface, but rather implementing the interface directly in the `RouteRefreshController`:
```
internal class RouteRefreshController(): Function0<CurrentIndicesSnapshot>
```
and passing the controller as the parameter for the wrapper (abstracted away) to avoid delegating this to the provider over which we have no control because any component can theoretically access it at any time.

Reply:
It's not very easy to do because of the initialization order in `MapboxNavigation`, but OK, makes sense, I'll look into it.